storage/src/adb/any/ext.rs

```diff
@@ -0,0 +1,186 @@
+            _ => unreachable!("ensure_dirty guarantees Dirty state"),
+        }
+    }
+}
+
+impl<A> StoreDeletable for AnyExt<A>
+where
+    A: CleanAny,
+{
+    async fn delete(&mut self, key: Self::Key) -> Result<bool, Self::Error> {
+        self.ensure_dirty();
+        match self.inner.as_mut().expect("wrapper should never be empty") {
+            State::Dirty(dirty) => DirtyAny::delete(dirty, key).await,
+            _ => unreachable!("ensure_dirty guarantees Dirty state"),
+        }
+    }
+}
+
+impl<A> StorePersistable for AnyExt<A>
+where
+    A: CleanAny,
+{
+    async fn commit(&mut self) -> Result<(), Self::Error> {
+        // Merkleize before commit
+        self.ensure_clean().await?;
+        match self.inner.as_mut().expect("wrapper should never be empty") {
+            State::Clean(clean) => clean.commit(None).await.map(|_| ()),
+            _ => unreachable!("ensure_clean guarantees Clean state"),
+        }
+    }
+
+    async fn destroy(mut self) -> Result<(), Self::Error> {
+        // Merkleize before destroy
+        self.ensure_clean().await?;
+        match self.inner.take().expect("wrapper should never be empty") {
+            State::Clean(clean) => clean.destroy().await,
+            _ => unreachable!("ensure_clean guarantees Clean state"),
+        }
+    }
+}
+
+impl<A> LogStore for AnyExt<A>
+where
+    A: CleanAny,
+{
+    type Value = <A as LogStore>::Value;
+
+    fn is_empty(&self) -> bool {
+        match self.inner.as_ref().expect("wrapper should never be empty") {
+            State::Clean(clean) => clean.is_empty(),
+            State::Dirty(dirty) => dirty.is_empty(),
+        }
+    }
+
+    fn op_count(&self) -> Location {
+        match self.inner.as_ref().expect("wrapper should never be empty") {
+            State::Clean(clean) => clean.op_count(),
+            State::Dirty(dirty) => dirty.op_count(),
+        }
+    }
+
+    fn inactivity_floor_loc(&self) -> Location {
+        match self.inner.as_ref().expect("wrapper should never be empty") {
+            State::Clean(clean) => clean.inactivity_floor_loc(),
+            State::Dirty(dirty) => dirty.inactivity_floor_loc(),
+        }
+    }
+
+    async fn get_metadata(&self) -> Result<Option<Self::Value>, Error> {
+        match self.inner.as_ref().expect("wrapper should never be empty") {
+            State::Clean(clean) => clean.get_metadata().await,
+            State::Dirty(dirty) => dirty.get_metadata().await,
+        }
+    }
+}
+
+impl<A> LogStorePrunable for AnyExt<A>
+where
+    A: CleanAny,
+{
+    async fn prune(&mut self, prune_loc: Location) -> Result<(), Error> {
+        // Merkleize before prune
+        self.ensure_clean().await?;
+        match self.inner.as_mut().expect("wrapper should never be empty") {
+            State::Clean(clean) => clean.prune(prune_loc).await,
+            _ => unreachable!("ensure_clean guarantees Clean state"),
+        }
+    }
+}
+
+impl<A> Batchable for AnyExt<A>
```

Comment:
I guess this would override the optimized implementations if any provided by the wrapped db?

Reply:
I would expect you'd see a performance regression for any::unordered::* in the generation benchmarks if so.